### PR TITLE
Rework on Helm management

### PR DESCRIPTION
**Pull request by guilhem**
- create a chart for nfd as dependency
  - I'm going to push it upstream or on its own repository
- simplify configuration variables
- fix CRD management using special helm directory
- use affinity instead of nodeSelector
- use chart namespace

### CODE REVIEW

**Comment by RenaudWasTaken** [comment]
Hello!

Thanks for your contribution! Can you:
- Move your PR to https://gitlab.com/nvidia/kubernetes/gpu-operator (the code is mirrored from there)
- Sign your work: https://help.github.com/en/github/authenticating-to-github/signing-commits

Thanks!

File: deployments/gpu-operator/values.yaml [hunk 1]
```diff
@@ -7,16 +7,52 @@ replicaCount: 1
...
+          - vendor
+
 platform:
   openshift: false
 
 operator:
   repository: nvidia
   image: gpu-operator
-  tag: @OPERATOR_VERSION@
+  tag: 1.0.0
```
Comment (RenaudWasTaken) [inline deployments/gpu-operator/values.yaml:34]: Can you leave this one to @OPERATOR_VERSION@, we need to change the CI to set it correctly before correcting this

File: deployments/gpu-operator/templates/serviceaccount.yaml [hunk 2]
```diff
@@ -2,4 +2,4 @@ apiVersion: v1
 kind: ServiceAccount
 metadata:
   name: special-resource-operator
-  namespace: gpu-operator
+  namespace: {{ $.Release.Namespace }}
```
Comment (RenaudWasTaken) [inline deployments/gpu-operator/templates/serviceaccount.yaml:5]: Just to be clear, this means the GPU Operator is by default installed in the default namespace unless the user specifies a namespace and uses `--create-namespace`

**Comment by RenaudWasTaken** [comment]
By the way this is a super cool contribution! Thanks a lot for your work!

File: deployments/gpu-operator/values.yaml (hunk 1 above)
Comment (guilhem) [reply deployments/gpu-operator/values.yaml:34]: Is there a way to change this? (CI that update chart.yaml for example)
It's really hard to use it with this notation as git

File: deployments/gpu-operator/templates/serviceaccount.yaml (hunk 2 above)
Comment (guilhem) [reply deployments/gpu-operator/templates/serviceaccount.yaml:5]: its released in current or specified `--namespace`.
It's the best practice for helm charts

**Comment by guilhem** [comment]
Thank you @RenaudWasTaken 
Done here: https://gitlab.com/nvidia/kubernetes/gpu-operator/-/merge_requests/80/

File: deployments/gpu-operator/values.yaml (hunk 1 above)
Comment (RenaudWasTaken) [reply deployments/gpu-operator/values.yaml:34]: Yes but let's do this in a followup PR

File: deployments/gpu-operator/templates/serviceaccount.yaml (hunk 2 above)
Comment (RenaudWasTaken) [reply deployments/gpu-operator/templates/serviceaccount.yaml:5]: Ok, I'll update the readme then

File: deployments/gpu-operator/values.yaml (hunk 1 above)
Comment (guilhem) [reply deployments/gpu-operator/values.yaml:34]: Yes of course :)
I update the PR on GitLab to remove the commit which change this.
(I anticipated your comment ;))

**Comment by RenaudWasTaken** [comment]
Closed with 2f2a708974d08ad6eefc439068d3ff309494b5cc

**Pull request closed without merging**